### ISSUE TITLE
Support DMN diagram accessibility 

### ISSUE THREAD

**Issue by ngohungphuc** [bpmn-io/dmn-js#778]
__Is your feature request related to a problem? Please describe.__

We are working on improving our accessibility by providing keyboard support that allowed user to be able to access the diagram using tab on the keyboard. Right now we don't see any kind of accessibility support for this library

__Describe the solution you'd like__

Support tab behaviour 

__Describe alternatives you've considered__

This could be worth checking https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard

__Additional context__

None

**Comment by barmac** [comment]
Hi,

thanks for opening this issue. Could you please point out the places where the library could be more accessible? How would you imagine keyboard navigation of a diagram modeling tool?

**Comment by nikku** [comment]
To add to @barmac comments: The editor is decently keyboard navigatable, allowing you to tab back and forth between cells.

![capture Dak0uh_optimized](https://github.com/bpmn-io/dmn-js/assets/58601/c5bcbcf0-b957-43f1-9f61-906a5bba74c3)

**Comment by ngohungphuc** [comment]
<img width="1004" alt="Scherm­afbeelding 2023-10-16 om 11 54 14" src="https://github.com/bpmn-io/dmn-js/assets/13591213/acaa352a-9da3-4a7b-9ab5-3bdb9c77e7c6">
@barmac Sorry for late reply. You can see on the demo they keyboard is not accessible you have to click on the blue icon to go to the detail view and then you can use accessible feature from there like @nikku mentioned.

**Comment by barmac** [comment]
Thanks for sharing. How would you expect to reach the desired element? I can imagine that we switch the selection via TAB, but what order should we follow? Or maybe there is another way?
How do I drill-down? My bet is via ENTER, but then we also need to think of a way to exit the decision.

**Comment by ngohungphuc** [comment]
@barmac I hope that we can switch the selection via TAB.

**Comment by ngohungphuc** [comment]
So any idea how we should tackle this issue ? @barmac

**Comment by BrianJVarley** [comment]
> I can see two ways to approach it:
> 
> 1. Make elements focusable and use browser focus handling to switch elements. Change dmn-js selection when element is focused.
> 2. Change dmn-js selection on `TAB` and `Shift+TAB` based on the order in `elementRegistry`.
> 
> If I were to tackle this topic, I'd probably try out one of the approaches, and also examine if it is accessible for keyboard-only users. Afterwards, I'd think about how to open and exit decision tables and literal expressions.

Hey @barmac, I'm working with @ngohungphuc on this and opened a PR here  - https://github.com/bpmn-io/dmn-js/pull/802 to try and address the keyboard focus for both _DrillDown_ and _ViewDrd_ buttons. Like you described in **point 2** we would like Tab key event to sequentially focus on each DrillDown button on the DMN Diagram and hitting Enter key will trigger the drill down. While vice versa focusing on the ViewDrd button and hitting Enter key will minimise the decision table.

Would be great to get your thoughts on the code changes. Also I have one unit test in progress as I've found it a bit tricky to mimic the enter event on the DrillDown button if you have some ideas?

Here is a short demo of the changes using MacOS Voiceover while performing those keyboard navigation steps:
![dmn-focus-btn-fix-demo](https://github.com/bpmn-io/dmn-js/assets/6731968/39418bc9-5b29-4f08-bd23-06282c53c3cb)
